handlers/twitter/twitter.go

```diff
@@ -0,0 +1,321 @@
+package twitter
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/dghubble/oauth1"
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/courier/utils"
+	"github.com/nyaruka/gocommon/urns"
+	"github.com/pkg/errors"
+)
+
+// Endpoints we hit
+var sendURL = "https://api.twitter.com/1.1/direct_messages/events/new.json"
+
+// DMs can be up to 10,000 characters long
+var maxMsgLength = 10000
+
+// Labels in quick replies can't be more than 36 characters
+var maxOptionLength = 36
+
+const (
+	configHandleID          = "handle_id"
+	configAPIKey            = "api_key"
+	configAPISecret         = "api_secret"
```

Comment:
configAPISecret is defined in channel.go

Reply:
Oh right, thanks for reminding me, going to remove it there.